app/controllers/registrations_controller.rb

```diff
@@ -17,10 +17,16 @@ def cancel_email_change!
   end
 
   def destroy
-    resource.discard
-    Devise.sign_out_all_scopes ? sign_out : sign_out(resource_name)
-    set_flash_message! :notice, :destroyed
-    respond_with_navigational(resource) { redirect_to after_sign_out_path_for(resource_name), status: Devise.responder.redirect_status }
+    if resource.valid_password?(user_delete_params[:current_password])
+      resource.user_account_feedbacks.create(user: resource, feedback: user_delete_params[:feedback]) if user_delete_params[:feedback]
```

Comment:
do we need an if condition here? we would create the record regardless of if there is feedback or not. the feedback can be nil but the record still needs to be made so we know when we did this to the user

Reply:
> do we need an if condition here? we would create the record regardless of if there is feedback or not. the feedback can be nil but the record still needs to be made so we know when we did this to the user

the condition is only to check if the password is valid